### PR TITLE
Fix bookkeeping of the field name

### PR DESCRIPTION
**Pull request by Neverlord**
The `field_` of `json_reader` keeps track of the nested field name in the JSON object. This PR fixes a bug where the `field_.pop_back();` had no prior `field_.push_back(...)` in some cases.

### CODE REVIEW

File: libcaf_core/src/json_reader.cpp [hunk 1]
```diff
@@ -297,13 +299,16 @@ bool json_reader::begin_field(std::string_view name,
...
+    // Always push since we'll do a .pop_back() in end_field().
+    field_.push_back(name);
     return false;
   }
 }
 
 bool json_reader::begin_field(std::string_view name, bool& is_present,
                               span<const type_id_t> types, size_t& index) {
   SCOPE(position::object);
+  field_.push_back(name);
```
Comment (josephnoir) [inline libcaf_core/src/json_reader.cpp:311]: In this function, `push_back(name)` can happen twice. Here and in line 320. Is that correct?

Comment (Neverlord) [reply libcaf_core/src/json_reader.cpp:311]: Oops. Fixed.

**Comment by Neverlord** [comment]
@josephnoir anything left to do here?

**Comment by josephnoir** [comment]
Don't think so.

**Pull request merged**